### PR TITLE
(std/node): add qs

### PR DESCRIPTION
**Pull request by zhmushan**
cc @kevinkassimo  ported from nodejs
#3403

### CODE REVIEW

File: std/node/querystring_test.ts [hunk 1]
```diff
@@ -0,0 +1,563 @@
+import * as qs from "./querystring.ts";
```
Comment (ry) [inline std/node/querystring_test.ts]: Please maintain original copyright headers

**Comment by justjavac** [comment]
Deno uses browser-compatible canonical `URL` and `searchParams`, and we don't have the burden of lower version compatibility. I think we should implement a more elegant query parsing library without using the implementation of node.

node's qs library basically reimplements the functionality of `searchParams`.

**Comment by zhmushan** [comment]
@justjavac Of course, but this module is for node compatibility.

File: std/node/_querystring.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN
+// NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+// OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
+// USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+export const hexTable = new Array<string>(256);
```
Comment (kevinkassimo) [inline std/node/_querystring.ts]: Maybe merge this file into `querystring.ts`? Seems odd to have a separate file here...

**Comment by justjavac** [comment]
node added whatwg URL in v6.13.0

**Comment by ry** [comment]
I think what @justjavac is suggesting is that we first implement the query string part of the URL spec in Deno. Then std/node/qs can use that implementation, rather than having a completely separate one.

**Comment by zhmushan** [comment]
@ry Hmm, but I need some free time to do this, maybe next month

**Comment by ry** [comment]
What if we land this PR but add a TODO to replace the implementation with URL.searchParams later? What do you think @justjavac ?

**Comment by justjavac** [comment]
@ry Please mark this PR as TODO. I will do it with @zhmushan

**Comment by sholladay** [comment]
As far as I know, `(new URL(url)).searchParams` can already be used here. I've been relying on it in [Pogo](https://GitHub.com/sholladay/pogo) without any known issues.

**Comment by bartlomieju** [comment]
@zhmushan @justjavac any update?

**Comment by bartlomieju** [comment]
Closing as stale. Feel free to reopen/open a new PR

**Pull request closed without merging**